src/site/content/en/blog/optimize-ttfb/index.md

```diff
@@ -0,0 +1,229 @@
+
+### Avoid multiple page redirects
+
+One potential contributor to a high TTFB is [redirects](https://developer.mozilla.org/docs/Web/HTTP/Redirections). Redirects occur when a navigation request for a document receives a response that informs the browser that the resource exists at another location. One redirect can certainly add unwanted latency to a navigation request, but it can certainly get worse if that redirect points to another resource that results in _another_ redirect—and so on.
+
+There are two types of redirects:
+
+- **Same-origin redirects**, where the redirect occurs entirely on your website.
+- **Cross-origin redirects**, where the redirect occurs initially on another origin—such as from a social media URL shortening service, for example—before arriving at your website.
+
+You want to focus on eliminating same-origin redirects, as this is something you will have direct control over. This would involve checking links on your website to see if any of them result in a 302 or 301 response code. When it comes to cross-origin redirects, there's nothing you can really do to fix that source of added latency.
+
+Another important source of redirect time can come from HTTP-to-HTTPS redirects. One way you can get around this is to use the [`Strict-Transport-Security` header](https://developer.mozilla.org/docs/Web/HTTP/Headers/Strict-Transport-Security) (HSTS), which will enforce HTTPS on the first visit to an origin, and then will tell the browser to immediately access the origin through the HTTPS scheme on future visits.
+
+Once you have a good HSTS policy in place, you can speed things up on the first visit to an origin [by adding your site to the HSTS preload list](https://developer.mozilla.org/docs/Web/HTTP/Headers/Strict-Transport-Security#preloading_strict_transport_security).
+
+{% Aside 'caution' %}
+Be **very careful** when implementing HSTS, as setting too aggressive a policy without sufficient testing [can break your website](/bbc-hsts/#deploying-hsts).
+{% endAside %}
+
+### Stream markup to the browser
+
+Browsers are optimized to deliver markup efficiently through streaming, meaning that markup is handled in chunks as it arrives from the server. This is crucial where large markup payloads are concerned, as it means the browser can parse that large chunk of markup incrementally, as opposed to waiting for the entire response to arrive before parsing can begin.
+
+Though browsers are great at streaming markup, it's crucial to do all that you can to keep that stream flowing so those initial bits of markup are on their way as soon as possible. If the back end is holding things up, that's a problem. Because back end stacks are numerous, it would be beyond the scope of this guide to cover every single stack and the issues that could arise in each specific one.
```

Comment:
```suggestion
Though browsers are great at handling streaming markup, it's crucial to do all that you can to keep that stream flowing so those initial bits of markup are on their way as soon as possible. If the backend is holding things up, that's a problem. Because backend stacks are numerous, it would be beyond the scope of this guide to cover every single stack and the issues that could arise in each specific one.
```